weather_mv/loader_pipeline/sinks.py

```diff
@@ -406,7 +409,19 @@ def open_dataset(uri: str,
     """Open the dataset at 'uri' and return a xarray.Dataset."""
     try:
         if is_zarr:
+            start_date = end_date = None
+            if 'start_date' in open_dataset_kwargs:
+                start_date = open_dataset_kwargs['start_date']
+                if open_dataset_kwargs is not None:
+                    open_dataset_kwargs.pop('start_date', None)
+            if 'end_date' in open_dataset_kwargs:
+                end_date = open_dataset_kwargs['end_date']
+                if open_dataset_kwargs is not None:
+                    open_dataset_kwargs.pop('end_date', None)
             ds: xr.Dataset = _add_is_normalized_attr(xr.open_dataset(uri, engine='zarr', **open_dataset_kwargs), False)
+            if start_date is not None and end_date is not None:
+                time_range = pd.date_range(start_date, end_date, periods=25, inclusive='left')
+                ds = ds.sel(time=time_range)
```

Comment:
Did you see my other comment? why not just pass `start_date` and `end_date` into the `sel` via slice? 
```suggestion
                ds = ds.sel(time=slice(start_date, end_date))
```

Reply:
Sorry I forgot the above comments. now it's changed.